TESTS/mbed_drivers/ticker/main.cpp

```diff
@@ -323,15 +323,81 @@ void test_attach_us_time(void)
     TEST_ASSERT_UINT64_WITHIN(TOLERANCE_US, DELAY_US, time_diff);
 }
 
+#if __cplusplus >= 201103 && !defined __CC_ARM
+/** Test single callback time via attach(Callback<void()> func, std::chrono::microseconds t)
+
+    Given a Ticker
+    When callback attached with time interval specified
+    Then ticker properly executes callback within a specified time interval
+ */
+template<us_timestamp_t DELAY_US>
+void test_attach_chrono_microseconds(void)
+{
+    Ticker ticker;
+    ticker_callback_flag = 0;
+
+    gtimer.reset();
+    gtimer.start();
+    ticker.attach(callback(stop_gtimer_set_flag), std::chrono::microseconds(DELAY_US));
+    while (!ticker_callback_flag);
+    ticker.detach();
+    const int time_diff = gtimer.read_us();
+
+    TEST_ASSERT_UINT64_WITHIN(TOLERANCE_US, DELAY_US, time_diff);
+}
+
+void test_attach_chrono_literals_10ms(void)
+{
+    Ticker ticker;
+    ticker_callback_flag = 0;
+    us_timestamp_t DELAY_US = 10000;
+
+    gtimer.reset();
+    gtimer.start();
+    ticker.attach(callback(stop_gtimer_set_flag), 10ms);
```

Comment:
I don't think this should be compiling, unless someone has done `using namespace std::chrono` or `using namespace std::chrono_literals`, which I don't see. The literals aren't available by default by just including `<chrono>`.

Any idea what's happening?

Reply:
I also thought that this would not compile as is but it did. Not sure why; I haven't spent time investigating.

Reply:
Ah, I get it. The default `using namespace std` from mbed.h makes them visible.

They're in the namespace `std::literals::chrono_literals`, with `literals` and `chrono_literals` both being `inline`, which means they appear as if in `std`.

(Ref: https://stackoverflow.com/questions/39065226/what-is-the-benefit-of-stdliterals-being-inline-namespaces )

Reply:
Thanks for looking into this.